src/client/components/app/Transactions/WithdrawTx.tsx

```diff
@@ -140,21 +142,32 @@ export const WithdrawTx: FC<WithdrawTxProps> = ({ header, onClose, children, ...
     walletNetwork,
   });
 
-  const selectedVaultOption = {
-    address: selectedVault.address,
-    symbol: selectedVault.displayName,
-    icon: selectedVault.displayIcon,
-    balance: selectedVault.DEPOSIT.userDeposited,
-    balanceUsdc: selectedVault.DEPOSIT.userDepositedUsdc,
-    decimals: selectedVault.token.decimals,
-  };
   const amountValue = toBN(amount).times(normalizeAmount(selectedVault.token.priceUsdc, USDC_DECIMALS)).toString();
   const expectedAmount = toBN(debouncedAmount).gt(0)
     ? normalizeAmount(expectedTxOutcome?.targetTokenAmount, selectedTargetToken?.decimals)
     : '';
   const expectedAmountValue = toBN(debouncedAmount).gt(0)
     ? normalizeAmount(expectedTxOutcome?.targetTokenAmountUsdc, USDC_DECIMALS)
     : '0';
+  const underlyingTokenBalance = calculateUnderlyingAmount({
+    amount: toUnit(selectedVault.DEPOSIT.userBalance, parseInt(selectedVault.decimals)),
+    decimals: selectedVault.token.decimals.toString(),
+    pricePerShare: selectedVault.pricePerShare,
+  });
+  const percentageToWithdraw = toBN(amount)
+    .div(toUnit(underlyingTokenBalance, selectedVault.token.decimals))
+    .times(100)
+    .toString();
+  const willWithdrawAll = toBN(percentageToWithdraw).gte(99);
```

Comment:
why don't we check for 100 directly?

Reply:
Price Per Share can slightly change in some cases, which leads to users having dust on their vault tokens.